internal/server/schema.sql

```diff
@@ -0,0 +1,168 @@
+-- SQLite schema for LBX photo metadata.
+
+PRAGMA foreign_keys = ON;
+
+------ Folders
+CREATE TABLE folders (
+    id INTEGER PRIMARY KEY ASC,
+    name TEXT NOT NULL,
+    path TEXT NOT NULL UNIQUE
+);
+CREATE UNIQUE INDEX folders_path ON folders(path);
+
+CREATE TABLE folders_tree (
+    parent_id INTEGER NOT NULL,
+    child_id INTEGER, -- NULL for media folders
+    FOREIGN KEY(parent_id) REFERENCES folders(id),
+    FOREIGN KEY(child_id) REFERENCES folders(id)
+);
+CREATE INDEX folders_tree_parent_id ON folders_tree(parent_id);
+CREATE INDEX folders_tree_child_id ON folders_tree(child_id);
+
+CREATE TABLE folders_albums (
+    folder_id INTEGER NOT NULL,
+    album_id INTEGER NOT NULL,
+    FOREIGN KEY(folder_id) REFERENCES folders(id),
+    FOREIGN KEY(album_id) REFERENCES albums(id)
+);
+CREATE INDEX folders_albums_folder_id ON folders_albums(folder_id);
+CREATE INDEX folders_albums_album_id ON folders_albums(album_id);
+
+------ Albums
+CREATE TABLE albums (
+    id INTEGER PRIMARY KEY ASC,
+    name TEXT NOT NULL, -- Display name in URL
+    path TEXT NOT NULL UNIQUE, -- Slash-separated folder path, minus name (e.g., "foo/bar/baz")
+    title_photo INTEGER REFERENCES media(id),
+    highlight_photo INTEGER REFERENCES media(id),
+    -- sort_order: 0: name, 1: name:rev, 2:mtime, 3:mtime:rev, 4:exif_time, 5:exif_time:rev
+    sort_order INTEGER NOT NULL DEFAULT 0
+);
+
+CREATE TABLE album_text (
+    album_id INTEGER NOT NULL,
+    language_code TEXT NOT NULL,
+    title TEXT NOT NULL,
+    blurb TEXT,
+    FOREIGN KEY(album_id) REFERENCES albums(id)
+);
+CREATE INDEX album_text_album_language ON album_text(album_id, language_code);
+
+------ Album path aliases. Opaque strings. Must be unique.
+CREATE TABLE album_aliases (
+    alias TEXT PRIMARY KEY,
+    album_id INTEGER NOT NULL,
+    FOREIGN KEY(album_id) REFERENCES albums(id)
+);
+CREATE INDEX album_aliases_album_id ON album_aliases(album_id);
+
+-- Media (photos / videos).
+-- Describes a logical photo, not a specific file (may have multiple resolutions).
+CREATE TABLE media (
+    id INTEGER PRIMARY KEY,
+    album_id INTEGER NOT NULL,
+    is_photo INTEGER NOT NULL, -- 0: video, 1: photo
+    display_name TEXT NOT NULL,
+    source_filename TEXT NOT NULL,
+    mtime INTEGER NOT NULL,
+    -- EXIF data
+    exif_time INTEGER,
+    latitude REAL,
+    longitude REAL,
+    camera TEXT,
+    lens TEXT,
+    focal_length REAL,
+    exposure_time REAL,
+    aperture REAL,
+    iso INTEGER,
+    flash INTEGER,
+    orientation INTEGER, -- 0: landscape, 1: portrait
+    FOREIGN KEY(album_id) REFERENCES albums(id)
+);
+CREATE INDEX media_album_id ON media(album_id);
+
+CREATE TABLE media_text (
```

Comment:
Can one `media` have multiple `media_text`s? What does that mean? If there's a 1:1 relationship, shouldn't these just be part of `media`?

Reply:
Yes, multiple media texts depending on language_code.

Reply:
Oh duh.